react/IntentModal/IntentModal.jsx

```diff
@@ -9,36 +9,59 @@ import PropTypes from 'prop-types'
  *
  * The modal for an intent takes the majority of the viewport.
  */
-const IntentModal = ({
-  closable,
-  overflowHidden,
-  dismissAction,
-  action,
-  doctype,
-  options,
-  onComplete,
-  onError,
-  create
-}) => (
-  <Modal
-    key='modal'
-    closable={closable}
-    overflowHidden
-    className={styles.intentModal}
-    crossClassName={styles.intentModal__cross}
-    dismissAction={dismissAction}
-  >
-    <IntentIframe
-      action={action}
-      create={create}
-      data={options}
-      onCancel={dismissAction}
-      onError={onError}
-      onTerminate={onComplete}
-      type={doctype}
-    />
-  </Modal>
-)
+class IntentModal extends Component {
+
+  // As dismissAction is passed twice to the modal, both for closing and for
+  // intent cancellation, we need to ensure that it is only actually
+  // called once.
+  // FIXME: this should be fixed by diferenciating dismissAction (for closing
+  // modal) and onCancel (for intent cancellation), but it implies deprecating
+  // dismissAction first, ensure legacy, prevent regressions, etc.
+  dismiss = () => {
+    const { status } = this.state
+    if (status === 'dismissed') return
+
+    const { dismissAction } = this.props
+    this.setState({
+      status: 'dismissed'
+    })
+    dismissAction && dismissAction()
+  }
```

Comment:
It would be better to use _.once for this usecase, it boils down to : 

```
const once = fn => {
    const called = false
    let result
	return function () {
      if (called) { return result }
      called = true
      result = fn.apply(this, arguments)
      return result
	}
}

dismiss = once(() => {
```

Reply:
@ptbrowne Well spotted, thanks !